### PR TITLE
APPS-1794 revert removing sys reqs args in DXExectuable run

### PR DESCRIPTION
**Pull request by YuxinShi0423**
This PR is to 

- add back system requirement related arguments `instance_type` and `cluster_spec` to DXExecutable.run
- add `fpga_driver` arg as well, which was missing from previous implementation

### CODE REVIEW

**Comment by commandlinegirl** [comment]
Approved modulo integration test runs pass. Thank you!

**Comment by r-i-v-a** [comment]
Could the PR also update the changelog file -- both re: the previously released version and the fix version? @YuxinShi0423 @commandlinegirl @kpjensen

File: src/python/test/test_dxclient.py [hunk 1]
```diff
@@ -3553,7 +3553,7 @@ def get_new_job_desc(cmd_suffix):
                           '*': {'instanceType': 'mem2_hdd2_v2_x2'}})
         check_new_job_metadata(new_job_desc, orig_job_desc, overridden_fields=['systemRequirements'])
 
-        def check_instance_count(job_desc:dict, entrypoints:list, expected_counts:list):
+        def check_instance_count(job_desc , entrypoints , expected_counts):
```
Comment (YuxinShi0423) [inline src/python/test/test_dxclient.py:3556]: fix invalid syntax in py2.7

File: src/python/dxpy/bindings/dxapplet.py [hunk 2]
```diff
@@ -58,6 +58,12 @@ def _get_run_input_common_fields(executable_input, **kwargs):
             if kwargs.get(arg) is not None:
                 run_input[arg] = kwargs[arg]
 
+        if kwargs.get('instance_type') is not None or kwargs.get('cluster_spec') is not None or kwargs.get('fpga_driver') is not None:
```
Comment (YuxinShi0423) [inline src/python/dxpy/bindings/dxapplet.py:61]: added back `instance_type` and `cluster_spec`

added `fpga_driver`

**Comment by r-i-v-a** [comment]
Ignore earlier comment about changelog since the code wasn't released yet.

**Review by kpjensen — CHANGES_REQUESTED** [review]
Have not reviewed code, but please reopen this PR in a new branch with your changes rebased off of the latest master to avoid any confusing history.

**Pull request closed without merging**